components/script/dom/globalscope.rs

```diff
@@ -131,6 +133,9 @@ pub struct GlobalScope {
     /// Vector storing closing references of all workers
     #[ignore_malloc_size_of = "Arc"]
     list_auto_close_worker: DomRefCell<Vec<AutoCloseWorker>>,
+
+    /// Vector storing references of all eventsources.
+    event_sources: DomRefCell<Vec<DomRoot<EventSource>>>,
```

Comment:
The event sources will now never be GCed while they're in this vector. Instead, if we store a WeakRefVec and add the weakReferenceable attribute to the EventSource interface in Bindings.conf, we can ensure that these references don't keep the underlying objects alive.

Reply:
Thanks! 

@KiChjang that seems that the answer we were looking for in https://github.com/servo/servo/pull/21218 as well...

Reply:
@gterzian Yes, I had a suspicion that DOM struct field members will not be GC'd, so we'd definitely need to pop the elements off of the Vec somehow...